core/trino-main/src/main/java/io/trino/SystemSessionProperties.java

```diff
@@ -220,6 +220,7 @@ public final class SystemSessionProperties
     public static final String COLUMNAR_FILTER_EVALUATION_ENABLED = "columnar_filter_evaluation_enabled";
     public static final String SPOOLING_ENABLED = "spooling_enabled";
     public static final String DEBUG_ADAPTIVE_PLANNER = "debug_adaptive_planner";
+    public static final String SOURCE_PAGES_VALIDATION_ENABLED = "output_pages_validation_enabled";
```

Comment:
> `Add output types validation to ScanFilterAndProjectOperator`

add color to the commit

```
Add output types validation to table scanning operators

We observed occurrences where Trino queries failed at runtime with ClassCastException
as pages produced by one of the Operators did not match the contract driven by query plan,
channels and their types. This adds some verifications aimed to help identify source of the problem

The idea is to verify pages returned by connectors. Ideally the verification would be applied
at the `ConnectorPageSource` level. However, it must not cause eager loading of data and
`ConnectorPageSource` returns source pages (`SourcePage`) which are lazy. Therefore
the validation is applied on `ScanFilterAndProjectOperator` and `TableScanOperator`.
```

Reply:
i see that TSO is separate commit (``Add output types validation to TableScanOperator``).
ok to keep separate, just update the proposed wording accordingly and mention that `TableScanOperator` will also be updated.